### PR TITLE
Add editorconfig

### PR DESCRIPTION
**Pull request by jaswilli**

### CODE REVIEW

**Comment by morficus** [comment]
I would suggest having 2-space indentation for the `package.json` files and all YAML files.
* YAML, because it is white-space sensitive
* package.json because when modifying package.json via npm (i.e.: `npm install`), it always treats package.json to be indented with 2-spaces

**Comment by jaswilli** [comment]
> YAML, because it is white-space sensitive

I don't think we're likely to have another YAML file in the project, which is why I only explicitly added an entry for `.travis.yml`, which we're currently using with a 2-space indentation.  I don't think YAML cares either way how much something is indented, just that indentation is uniform within the file.

That said, I wouldn't have an issue changing the rule so that all YAML files are 2-space.


> package.json because when modifying package.json via npm (i.e.: npm install), it always treats package.json to be indented with 2-spaces

Our `package.json` is actually 4-space indented and I think it's pretty much only edited by hand because npm is always changing the spacing and ordering on us and it wreaks havoc on the diffs.  

I did forget about `core/client/package.json` though.  That needs its own rule because it's managed by ember-cli which only does 2-space indentation.  Same with `core/client/bower.json` for the same reason.

So tl;dr, I'll update this to make all YAML files 2-space indentation to match `.travis.yml` and add separate rules for the ember-cli managed `bower.json` and `package.json` files.

**Comment by ErisDS** [comment]
This is working really well for me with intellij & saving some headaches :+1: 

Seems like this would be a good point at which to switch from the 4 spaces in our root package.json to the standard 2. We only used 4 in order to have project-wide consistency which is kind of a mute point now, and this seems like the right PR to make that change?

**Comment by jaswilli** [comment]
Sounds alright to me.  Updated again with a re-indented `package.json`.

**Comment by rwjblue** [comment]
:heart:
